node/rpc/src/chain_api.rs

```diff
@@ -55,6 +58,18 @@ where
     DB: Blockstore + Store + Clone + Send + Sync + 'static,
     B: Beacon,
 {
+    lazy_static::lazy_static! {
+        static ref LOCK: Mutex<()> = Mutex::new(());
+    }
+
+    let _locked = LOCK.try_lock();
+    if _locked.is_err() {
+        return Err(JsonRpcError::Provided {
+            code: -100,
```

Comment:
It would be nice to use standard error codes
```suggestion
            code: http::StatusCode::SERVICE_UNAVAILABLE,
```

Reply:
Fixed